stdlib/Markdown/src/Julia/interp.jl

```diff
@@ -11,7 +11,7 @@ function interpinner(stream::IO, greedy = false)
     startswith(stream, '$') || return
     (eof(stream) || peek(stream, Char) in whitespace) && return
     try
-        return _parse(stream::IOBuffer, greedy = greedy)
+        return _parse(stream, greedy = greedy)
```

Comment:
This type-assert in `try-catch` threw sometimes since in some cases this function is passed a `GenericIOBuffer` instead of an `IOBuffer`....

Reply:
````
julia> using Markdown, MethodAnalysis

julia> methodinstances(Markdown.interpinner)
2-element Vector{Core.MethodInstance}:
 MethodInstance for interpinner(::IOBuffer)
 MethodInstance for interpinner(::IOBuffer, ::Bool)

julia> begin
           x = "X"
           contains_X(md) = occursin(x, sprint(show, MIME("text/plain"), md))
           @assert contains_X(md"# $x") # H1
           @assert contains_X(md"## $x") # H2
           @assert contains_X(md"### $x") # H3
           @assert contains_X(md"x = $x") # Paragraph
           @assert contains_X(md"- $x") # List
           @assert contains_X(md"[$x](..)") # Link
           @assert contains_X(md"**$x**") # Bold
           @assert contains_X(md"*$x*") # Italic
           @assert contains_X( # Table
               md"""
               | name |
               |------|
               |  $x  |
               """)
       end

julia> methodinstances(Markdown.interpinner)
4-element Vector{Core.MethodInstance}:
 MethodInstance for interpinner(::IOBuffer)
 MethodInstance for interpinner(::Base.GenericIOBuffer{SubArray{UInt8,1,Vector{UInt8},Tuple{UnitRange{Int64}},true}})
 MethodInstance for interpinner(::IOBuffer, ::Bool)
 MethodInstance for interpinner(::Base.GenericIOBuffer{SubArray{UInt8,1,Vector{UInt8},Tuple{UnitRange{Int64}},true}}, ::Bool)
````